File: src/za/co/adhd_developers/java/questions/Question7.java
package za.co.adhd_developers.java.questions;

import za.co.adhd_developers.java.tools.Question;
import za.co.adhd_developers.java.tools.Utils;

/**
 * Created by Grant on 2017/07/28.
 */
public class Question7 implements Question {

    int limit = 10001;
    int answer = 0;

    @Override
    public void doWork() {
        int count = 0;
        int currNumber = 2;

        while (count != limit) {
            if (Utils.newIsPrime(currNumber)) {
                count++;
            }

            if (count != limit) {
                currNumber++;
            }
        }

        this.answer = currNumber;
    }

    @Override
    public void printAnswer() {
        System.out.println("=================================");
        System.out.println("Question 7");
        System.out.println("The 10 001st prime number: " + this.answer);
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question7.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question
import za.co.adhd_developers.kotlin.tools.getNextPrime

class Question7 : Question {

    var answer = 0L

    override fun doWork() {
        var currentPrime = 2L

        for (i in 2..10001) {
            currentPrime = getNextPrime(currentPrime)
        }

        this.answer = currentPrime
    }

    override fun printAnswer() {
        println("=================================")
        println("Question 7")
        println("The 10 001st prime number: " + this.answer)
        println("=================================")
    }
}